subiquity/controllers/filesystem.py

```diff
@@ -85,59 +77,49 @@ def finish(self):
         # switch to next screen
         self.signal.emit_signal('next-screen')
 
-    # Filesystem/Disk partition -----------------------------------------------
-    def partition_disk(self, disk):
-        log.debug("In disk partition view, using "
-                  "{} as the disk.".format(disk.label))
-        dp_view = DiskPartitionView(self.model, self, disk)
+    def create_mount(self, fs, spec):
+        if spec['mount'] is None:
+            return
+        mount = self.model.add_mount(fs, spec['mount'])
+        return mount
+
+    def delete_mount(self, mount):
+        if mount is None:
+            return
+        self.model.remove_mount(mount)
 
-        self.ui.set_body(dp_view)
+    def create_filesystem(self, volume, spec):
+        if spec['fstype'] is None or spec['fstype'].label is None:
+            return
+        fs = self.model.add_filesystem(volume, spec['fstype'].label)
+        self.create_mount(fs, spec)
+        return fs
 
-    def add_disk_partition(self, disk):
-        log.debug("Adding partition to {}".format(disk))
-        adp_view = PartitionView(self.model, self, disk)
-        self.ui.set_body(adp_view)
+    def delete_filesystem(self, fs):
+        if fs is None:
+            return
+        self.delete_mount(fs.mount())
+        self.model.remove_filesystem(fs)
 
-    def edit_partition(self, disk, partition):
-        log.debug("Editing partition {}".format(partition))
-        adp_view = PartitionView(self.model, self, disk, partition)
-        self.ui.set_body(adp_view)
+    def create_partition(self, device, spec, flag=""):
```

Comment:
create_filesystem? and partition is implicit or do we always create them together?  What if we create a partition but leave it unformated?

Reply:
The spec argument contains fields to determine whether to create the filesystem or not. I'm not super happy about this bit of API design, but it's not actually very different from what is in master already.